### PR TITLE
Additional fixes related to output serialization and usability

### PR DESCRIPTION
**Pull request by kheki4**
- Temporal search loop was fixed to be left-inclusive and right-exclusive in RecEventTracker.cpp. 
- Also fixed message box blocking image recording start. 
- Also added option to general settings dialog to let the user choose csv output data structure (this will only affect data writer output, streaming is by default in accordance to the new data structure that includes trial numbering). 
- Also added separator line to File menu to prevent accidental clicks on Exit button.

### CODE REVIEW

File: src/subwindows/generalSettingsDialog.cpp [hunk 1]
```diff
@@ -95,37 +101,42 @@ void GeneralSettingsDialog::updateForm() {
...
+    //dataWriterDelimiterBox->setCurrentText(delimiterToUse);
+    if(dataWriterDelimiter == ";")
+        dataWriterDelimiterBox->setCurrentIndex(1);
+    else if(dataWriterDelimiter == "\t")
+        dataWriterDelimiterBox->setCurrentIndex(2);
+    else //if(dataWriterDelimiter == ",")
+        dataWriterDelimiterBox->setCurrentIndex(0);
+    qDebug() << "Data writer delimiter read as: " << dataWriterDelimiter << "\n";
+
+    if(dataWriterDataStructure == "PupilEXT-0-1-1")
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:130]: This should maybe use the index instead of the string?

File: src/recEventTracker.cpp [hunk 2]
```diff
@@ -341,8 +342,8 @@ void RecEventTracker::saveOfflineEventLog(uint64 timestampFrom, uint64 timestamp
 // below is the version that creates a csv
 /*
 void RecEventTracker::saveOfflineEventLog(uint64 timestampFrom, uint64 timestampTo, const QString& fileName) {
-    delim = applicationSettings->value("delimiterToUse", ",").toString()[0];
-    //delim = applicationSettings->value("delimiterToUse", ',').toChar(); // somehow this just doesnt work
+    delim = applicationSettings->value("dataWriterDelimiterToUse", ",").toString()[0];
```
Comment (sasfog) [inline src/recEventTracker.cpp:345]: This is commented out but I believe the variable is DataWriterDelimiter, not dataWriterDelimiterToUse. Probably should just delete the function if it's commented out anyway

File: src/subwindows/generalSettingsDialog.cpp [hunk 3]
```diff
@@ -156,13 +166,23 @@ void GeneralSettingsDialog::createForm() {
...
 
-    QLabel *delimiterLabel = new QLabel(tr("Delimiter Character"));
-    delimiterBox = new QComboBox();
-    delimiterBox->addItem(QString("Comma [,]"), QString(","));
-    delimiterBox->addItem(QString("Semicolon [;]"), QString(";"));
-    delimiterBox->addItem(QString("Tabulation"), QString("\t"));
-    delimiterBox->setCurrentText(delimiterToUse);
-    dataOutLayout->addRow(delimiterLabel, delimiterBox);
+    QLabel *dataWriterDelimiterLabel = new QLabel(tr("Delimiter Character"));
+    dataWriterDelimiterBox = new QComboBox();
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:170]: Add parent to avoid memory leaks

File: src/subwindows/generalSettingsDialog.cpp [hunk 4]
```diff
@@ -156,13 +166,23 @@ void GeneralSettingsDialog::createForm() {
...
     QFormLayout *dataOutLayout = new QFormLayout;
 
-    QLabel *delimiterLabel = new QLabel(tr("Delimiter Character"));
-    delimiterBox = new QComboBox();
-    delimiterBox->addItem(QString("Comma [,]"), QString(","));
-    delimiterBox->addItem(QString("Semicolon [;]"), QString(";"));
-    delimiterBox->addItem(QString("Tabulation"), QString("\t"));
-    delimiterBox->setCurrentText(delimiterToUse);
-    dataOutLayout->addRow(delimiterLabel, delimiterBox);
+    QLabel *dataWriterDelimiterLabel = new QLabel(tr("Delimiter Character"));
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:169]: Add parent to avoid memory leaks

File: src/subwindows/generalSettingsDialog.cpp [hunk 5]
```diff
@@ -156,13 +166,23 @@ void GeneralSettingsDialog::createForm() {
...
+    dataWriterDelimiterBox = new QComboBox();
+    dataWriterDelimiterBox->addItem(QString("Comma [,]"), QString(","));
+    dataWriterDelimiterBox->addItem(QString("Semicolon [;]"), QString(";"));
+    dataWriterDelimiterBox->addItem(QString("Tabulation"), QString("\t"));
+    dataWriterDelimiterBox->setCurrentText(dataWriterDelimiter);
+    dataOutLayout->addRow(dataWriterDelimiterLabel, dataWriterDelimiterBox);
+
+    QLabel *dataWriterDataStructureLabel = new QLabel(tr("Data Structure*: "));
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:177]: Add parent to avoid memory leaks

File: src/subwindows/generalSettingsDialog.cpp [hunk 6]
```diff
@@ -156,13 +166,23 @@ void GeneralSettingsDialog::createForm() {
...
+    dataWriterDataStructureBox = new QComboBox();
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:178]: Add parent to avoid memory leaks

File: src/subwindows/generalSettingsDialog.cpp [hunk 7]
```diff
@@ -156,13 +166,23 @@ void GeneralSettingsDialog::createForm() {
...
+    dataWriterDataStructureBox->addItem(QString("PupilEXT v0.1.1"), QString("PupilEXT-0-1-1"));
+    dataWriterDataStructureBox->addItem(QString("PupilEXT v0.1.2"), QString("PupilEXT-0-1-2"));
+    dataWriterDataStructureBox->setCurrentText(dataWriterDataStructure);
+    dataOutLayout->addRow(dataWriterDataStructureLabel, dataWriterDataStructureBox);
+    QLabel *dataWriterDataStructureWarnLabel = new QLabel(tr("*Older version will not save trial numbering."));
```
Comment (sasfog) [inline src/subwindows/generalSettingsDialog.cpp:183]: Add parent to avoid memory leaks

**Review by sasfog — CHANGES_REQUESTED** [review]
Please see comments

File: src/subwindows/generalSettingsDialog.cpp (hunk 5 above)
Comment (kheki4) [reply src/subwindows/generalSettingsDialog.cpp:177]: Done

File: src/subwindows/generalSettingsDialog.cpp (hunk 7 above)
Comment (kheki4) [reply src/subwindows/generalSettingsDialog.cpp:183]: Done

File: src/subwindows/generalSettingsDialog.h [hunk 8]
```diff
@@ -62,9 +63,11 @@ class GeneralSettingsDialog : public QDialog {
     bool metaSnapshotsEnabled;
     bool saveOfflineEventLog; 
 
-    QString delimiterToUse;
+    QString dataWriterDelimiter;
+    QString dataWriterDataStructure;
```
[the remark this answers is not in the document]
Comment (kheki4) [reply src/subwindows/generalSettingsDialog.h:67]: Done

File: src/subwindows/generalSettingsDialog.cpp (hunk 3 above)
Comment (kheki4) [reply src/subwindows/generalSettingsDialog.cpp:170]: Done

File: src/recEventTracker.cpp (hunk 2 above)
Comment (kheki4) [reply src/recEventTracker.cpp:345]: Done

File: src/subwindows/generalSettingsDialog.cpp (hunk 1 above)
Comment (kheki4) [reply src/subwindows/generalSettingsDialog.cpp:130]: Left like this for now

**Review by sasfog — APPROVED** [review]

**Pull request merged**